### PR TITLE
feat: Redirect users to the main page upon successful login

### PR DESCRIPTION
**Pull request by AbdugafurovaUmida**
1. Task: [link](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint2/RSS-ECOMM-2_05.md)

2. Screenshot: 
image

3. Deploy: [link](https://github.com/NataliaIv90/ecommerce-app)

4. Done 16.05.2024 / deadline 22.05.2024

5. Score: 15/15

Redirect users to the application's main page upon successful login, allowing them to access the main features and content of the application. The main page can contain simple text (e.g., "Main") or any other content to demonstrate that the user is successfully logged in. Ensure that the URL in the address bar is changed to the main page's URL, properly handling navigation and user experience during redirection.

## Description
URL Change and Redirection Tasks 🔀
Implement a function or module in your application that handles successful authentication by redirecting users from the login page to the main page.
Update the URL in the address bar to reflect the correct URL for the main page (e.g., "/main") upon successful login and redirection.
Ensure that the browser's navigation (history) is updated correctly when redirecting users to the main page, allowing them to use the back button to navigate to the previous page if necessary.
Make sure that the user's authentication state is preserved during the redirection process, allowing them to stay logged in when accessing the main page.
Acceptance Criteria ✅
After successful authentication, users are redirected to the main page of the application, which contains simple text ("Main") or any other content.
The URL in the address bar is changed to the main page's URL upon redirection.
Proper handling of navigation and user experience during redirection, including browser history updates and preserving authentication state.

### CODE REVIEW

File: src/hooks/ProtectRoutes.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React, { ReactNode } from 'react';
+import { useAuth } from './AuthHooks';
+
+interface ProtectRoutesProps {
+  children: ReactNode;
+}
+
+export const ProtectRoutes: React.FC<ProtectRoutesProps> = ({ children }) => {
+  useAuth();
```
Comment (MariamKochadze) [inline src/hooks/ProtectRoutes.tsx:9]: After useAuth() we should check if customer is logged in and if not we should navigate it to log in page maybe, something like this:  if (!isAuth) {
    return <Navigate to="/login" replace />; 
  }

File: src/components/main/Main.tsx [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
-    </>
-  </MainWrapper>
-);
+export const Main: React.FC = () => {
+  const customer = useAppSelector((state) => state.customers.customer);
+  return (
+    <MainWrapper>
+      <>
+        <p>Main</p>
+        <h3>Hello {customer.firstName} !</h3>
```
Comment (NataliaIv90) [inline src/components/main/Main.tsx:11]: Everything is working ✨

File: src/components/login/loginForm/LoginForm.tsx [hunk 3]
```diff
@@ -120,7 +123,7 @@ export const LoginForm = (): JSX.Element => {
                 error={
                   emailError ? 'Customer account with the given credentials not found.' : fieldState.error?.message
                 }
-                value={field.value}
+                value={customer ? customer.email : field.value}
```
Comment (NataliaIv90) [inline src/components/login/loginForm/LoginForm.tsx]: I'm not sure we have to use customer.email
It is better 
`value={field.value}`

File: src/components/login/loginForm/LoginForm.tsx [hunk 4]
```diff
@@ -139,7 +142,7 @@ export const LoginForm = (): JSX.Element => {
                 error={
                   passwordError ? 'Customer account with the given credentials not found.' : fieldState.error?.message
                 }
-                value={field.value}
+                value={customer ? customer.password : field.value}
```
Comment (NataliaIv90) [inline src/components/login/loginForm/LoginForm.tsx]: the same

**Review by NataliaIv90 — APPROVED** [review]

**Review by MariamKochadze — APPROVED** [review]

File: src/hooks/ProtectRoutes.tsx (hunk 1 above)
Comment (AbdugafurovaUmida) [reply src/hooks/ProtectRoutes.tsx:9]: The offer you refer to is checked in the custom hook

File: src/components/login/loginForm/LoginForm.tsx (hunk 4 above)
Comment (AbdugafurovaUmida) [reply src/components/login/loginForm/LoginForm.tsx]: fixed

File: src/components/login/loginForm/LoginForm.tsx (hunk 3 above)
Comment (AbdugafurovaUmida) [reply src/components/login/loginForm/LoginForm.tsx]: fixed

**Pull request merged**